solr/core/src/java/org/apache/solr/handler/RestoreCore.java

```diff
@@ -99,44 +102,47 @@ public boolean doRestore() throws Exception {
                   DirectoryFactory.DirContext.DEFAULT,
                   core.getSolrConfig().indexConfig.lockType);
 
-      // Prefer local copy.
-      indexDir =
-          core.getDirectoryFactory()
-              .get(
-                  indexDirPath,
-                  DirectoryFactory.DirContext.DEFAULT,
-                  core.getSolrConfig().indexConfig.lockType);
-      Set<String> indexDirFiles = new HashSet<>(Arrays.asList(indexDir.listAll()));
-      // Move all files from backupDir to restoreIndexDir
-      for (String filename : repository.listAllFiles()) {
-        checkInterrupted();
-        try {
-          if (indexDirFiles.contains(filename)) {
-            Checksum cs = repository.checksum(filename);
-            IndexFetcher.CompareResult compareResult;
-            if (cs == null) {
-              compareResult = new IndexFetcher.CompareResult();
-              compareResult.equal = false;
+      try (Lock lock = restoreIndexDir.obtainLock(IndexWriter.WRITE_LOCK_NAME)) {
```

Comment:
What this change does?

Reply:
same as https://github.com/cowpaths/fullstory-solr/pull/182/files#r1508329085